ecl/hql/hqltrans.cpp

```diff
@@ -4234,15 +4235,30 @@ IHqlExpression * ScopedTransformer::getEvaluateScope(IHqlExpression * scope)
 }
 
 
-void ScopedTransformer::pushScope()
+void ScopedTransformer::pushScope(IHqlExpression * context)
 {
-    innerScope = new ScopeInfo;
+    innerScope = new ScopeInfo(context);
     scopeStack.append(*innerScope);
+#ifdef TRACE_DEEP_SCOPES
+    static unsigned maxDepth = 8;
+    if (scopeStack.ordinality() > maxDepth)
+    {
+        printf("----- nesting -----\n");
+        ForEachItemIn(i, scopeStack)
+        {
+            ScopeInfo & cur = scopeStack.item(i);
+            StringBuffer temp;
+            toUserECL(temp, cur.context, false);
+            printf("%p: %s", cur.dataset, temp.str());
+        }
+        maxDepth = scopeStack.ordinality();
```

Comment:
@shamser so they I only get reports when the stack gets even deeper.  I'm interested in what the mostly deeply nested instance is - so once I have an example at one level, I only want to know if there is a deeper once.